### PR TITLE
test/ML-377-enable-tests

### PR DESCRIPTION
**Pull request by iryna-melnyk**
tests in ml_model_tests were partially enabled
I hope this PR will be merged after  2237 & 2243

### CODE REVIEW

File: libs/ml/include/ml/core/node.hpp [hunk 1]
```diff
@@ -237,8 +237,10 @@ std::shared_ptr<TensorType> Node<TensorType>::Evaluate(bool is_training)
     {
       throw std::runtime_error("NaN encountered in Node::Evaluate");
     }
-
-    assert(!math::state_overflow<DataType>());
+    if (math::state_overflow<DataType>())
```
Comment (baykaner) [inline libs/ml/include/ml/core/node.hpp]: I'm not convinced that we want to be doing this. What is the justification for switching to runtime overflow checks?

**Review by baykaner — CHANGES_REQUESTED** [review]
testing looks good. Not sure about the runtime node overflow checks though

File: libs/ml/include/ml/core/node.hpp (hunk 1 above)
Comment (iryna-melnyk) [reply libs/ml/include/ml/core/node.hpp]: My changes were based on  #2237 to avoid a lot of conflicts in ml_model_tests
I'll restore node.hpp

Comment (iryna-melnyk) [reply libs/ml/include/ml/core/node.hpp]: this branch was rebased so all unneeded changes are gone

**Review by baykaner — APPROVED** [review]

**Pull request merged**